src/Server.cpp

```diff
@@ -0,0 +1,57 @@
+#include "Server.hpp"
+#include "nlohmann/json.hpp"
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include "curl/curl.h"
+#include "Utility.hpp"
+#include "User.hpp"
+
+std::string data;
+
+size_t write_data(char* buf, size_t size, size_t nmemb, void* up)
+{
+    for (size_t c = 0; c < size*nmemb; c++)
+        data.push_back(buf[c]);
+    return size * nmemb;
+}
+
+void Server::downloadUsers()
+{
+    CURL* curl;
+    curl_global_init(CURL_GLOBAL_ALL);
+    curl = curl_easy_init();
+
+    if (curl)
+    {
+        std::string path = serverName + pathToFile;
+        curl_easy_setopt(curl, CURLOPT_URL, path.c_str());
+        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, &write_data);
+        curl_easy_perform(curl);
+
+        if (checkIfFileExist())
+        {
+            Users listOfUsers = convertToArray(data);
+            std::cout << "\nUsers has been uploaded into database ;) \n\n";
+            for (auto user : listOfUsers)
+                std::cout << user.getNick() << "\n";
+        }
+  
+        curl_easy_cleanup(curl);
+        curl_global_cleanup();
+    }
+    else
+        std::cout << "Somethng goes wrong. \n";
+}
+
+bool Server::checkIfFileExist()
+{
+    if (data[0] != '<')
```

Comment:
why `data[0] != '<'` means that the file exists?

Reply:
First character will be found '<' when file isn't exist on server. This is begin of HTML language with page 404 for example. So when we get sign '[' on first position of string we are sure that file is exist, something else on first position - file isn't exist. Path to file is wrong.  That was the simplest solution without getting involved in sockets etc. ;) -> huge shortcut

Reply:
And what if we want do have an xml or html file type instead of json? :) They both starts with '<'.
There must be another way to check if the file exist.